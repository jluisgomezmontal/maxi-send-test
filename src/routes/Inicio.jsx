import { Link } from "react-router-dom";
import placeholder from "../assets/placeholder.svg";

export const Inicio = () => {
  return (
    <div className="container-xl d-flex pt-1 gap-5 ">
      <Link to="series" className="text-decoration-none text-black">
        <div className="icon position-relative">
          <img
            className=" position-absolute top-50 start-50 translate-middle h-75"
            src={placeholder}
            alt="placeholder"
          />
          <h2 className="text-center text-white position-absolute top-50 start-50 translate-middle">
            SERIES
          </h2>
        </div>
        <h6>Popular Series</h6>
      </Link>
      <Link to="movies" className="text-decoration-none text-black">
        <div className="icon position-relative">
          <img
            className="position-absolute top-50 start-50 translate-middle h-75"
            src={placeholder}
            alt="placeholder"
          />
          <h2 className="text-center text-white position-absolute top-50 start-50 translate-middle">
            MOVIES
          </h2>
        </div>
        <h6>Popular Movies</h6>
      </Link>
    </div>
  );
};
