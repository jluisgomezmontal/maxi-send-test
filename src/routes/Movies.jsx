import { useState } from "react";
import { useData } from "../hooks/useData";
import { Modal } from "../components/Modal/Modal";
import { useDispatch, useSelector } from "react-redux";
import { setDataModal } from "../redux/slices/dataSlice";
import { FilterBTN } from "../components/FilterBTN/FilterBTN";

export const Movies = () => {
  const [filterByYear, setFilterByYear] = useState(false);
  const [modal, setModal] = useState(false);
  const [results, setResults] = useState(20);

  const [movies] = useData("movie", filterByYear, results);
  const { dataModal } = useSelector((state) => state.data);
  const dispatch = useDispatch();

  const [loadedImages, setLoadedImages] = useState(true);
  setTimeout(() => {
    setLoadedImages(false);
  }, 1000);
  return (
    <div className="container-xl">
      <FilterBTN setFilterByYear={setFilterByYear} setResults={setResults} />

      <div className="row gap-1">
        {loadedImages ? (
          <h5>Loading...</h5>
        ) : (
          movies.map((entrie, index) => (
            <div className="entrie_container text-white" key={index}>
              <img
                onClick={() => {
                  dispatch(setDataModal({ entrie }));
                  setModal(true);
                }}
                src={entrie.images.PosterArt.url}
                alt={entrie.title}
              />
              <p>{entrie.title}</p>
            </div>
          ))
        )}
        {modal && (
          <Modal
            dataModal={dataModal.entrie}
            setModal={setModal}
            modal={modal}
          />
        )}
      </div>
    </div>
  );
};
