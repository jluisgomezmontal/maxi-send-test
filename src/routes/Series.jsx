import { useState } from "react";
import { Link } from "react-router-dom";
import { Modal } from "../components/Modal/Modal";
import { useData } from "../hooks/useData";
import { setDataModal } from "../redux/slices/dataSlice";
import { useDispatch, useSelector } from "react-redux";

export const Series = () => {
  const [loadedImages, setLoadedImages] = useState(true);
  const [modal, setModal] = useState(false);
  const [series] = useData("series");

  const { dataModal } = useSelector((state) => state.data);
  const dispatch = useDispatch();

  setTimeout(() => {
    setLoadedImages(false);
  }, 1000);

  return (
    <div className="container-xl">
      <div className="row gap-1">
        {loadedImages ? (
          <h5>Loading...</h5>
        ) : (
          series.map((entrie, index) => (
            <Link className="entrie_container text-white" key={index}>
              <img
                onClick={() => {
                  dispatch(setDataModal({ entrie }));
                  setModal(true);
                }}
                src={entrie.images.PosterArt.url}
                alt={entrie.title}
                width={entrie.images.PosterArt.with}
              />
              <p>{entrie.title}</p>
            </Link>
          ))
        )}
        {modal && (
          <Modal
            dataModal={dataModal.entrie}
            setModal={setModal}
            modal={modal}
          />
        )}
      </div>
    </div>
  );
};
