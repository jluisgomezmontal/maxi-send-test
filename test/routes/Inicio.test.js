/**
 * @jest-environment jsdom
 */
/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import { Inicio } from "../../src/routes/Inicio";

describe("testing <Inicio/>", () => {
  test("should render propertly", () => {
    const { container } = render(
      <BrowserRouter>
        <Inicio />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test("should have series and movies", () => {
    render(
      <BrowserRouter>
        <Inicio />
      </BrowserRouter>
    );

    const seriesLink = screen.getByText("SERIES");
    const moviesLink = screen.getByText("MOVIES");

    expect(seriesLink).toBeInTheDocument();
    expect(moviesLink).toBeInTheDocument();
  });
});
