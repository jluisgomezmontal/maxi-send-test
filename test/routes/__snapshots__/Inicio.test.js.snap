// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Inicio/> should render propertly 1`] = `
<div>
  <div
    class="container-xl d-flex pt-1 gap-5 "
  >
    <a
      class="text-decoration-none text-black"
      href="/series"
    >
      <div
        class="icon position-relative"
      >
        <img
          alt="placeholder"
          class=" position-absolute top-50 start-50 translate-middle h-75"
        />
        <h2
          class="text-center text-white position-absolute top-50 start-50 translate-middle"
        >
          SERIES
        </h2>
      </div>
      <h6>
        Popular Series
      </h6>
    </a>
    <a
      class="text-decoration-none text-black"
      href="/movies"
    >
      <div
        class="icon position-relative"
      >
        <img
          alt="placeholder"
          class="position-absolute top-50 start-50 translate-middle h-75"
        />
        <h2
          class="text-center text-white position-absolute top-50 start-50 translate-middle"
        >
          MOVIES
        </h2>
      </div>
      <h6>
        Popular Movies
      </h6>
    </a>
  </div>
</div>
`;
